File: pxt-steami/libs/leds/leds.cpp
enum class AllLeds {
    //% block="LED_CHG"
    LED_CHG = 1,
    //% block="LED_STATUS"
    LED_STATUS = 2,
    //% block="LED_BLE"
    LED_BLE = 3,
    //% block="LED_FLASH"
    LED_FLASH = 4,
    //% block="LED_DAPLINK"
    LED_DAPLINK = 5,
    //% block="LED_RGB"
    LED_RGB = 6,
};

enum class AllColors {
    //% block="Red" color=#ff0000
    Red,
    //% block="Green" color=#00ff00
    Green,
    //% block="Blue" color=#0000ff
    Blue,
    //% block="Yellow" color=#ffff00
    Yellow,
    //% block="Magenta" color=#ff00ff
    Magenta,
    //% block="Cyan" color=#00ffff
    Cyan,
    //% block="White" color=#ffffff
    White,
};

//% block="leds" weight=100 color=#4287f5 icon=""
namespace leds {
/**
 * @brief Set LED on or off
 * @param led Led to set
 * @param on Turn on or off
 * @param color Turn on or off
 */
//% block="set led %led|to %on" weight=100
//% help=leds/leds_setLed
//% blockId="leds_setLed" blockGap=8
//% on.shadow="toggleOnOff" on.defl=true
void setLed(AllLeds led, bool on) {
    // to do
}

/**
 * @brief Set all LEDs on or off
 * @param on Turn on or off
 */
//% block="set all leds to %on" weight=90
//% help=leds/leds_setAllLeds
//% blockId="leds_setAllLeds" blockGap=8
//% on.shadow="toggleOnOff" on.defl=true
void setAllLeds(bool on) {
    // to do
}

/**
 * @brief Set LED RGB to color
 * @param color Color to set
 */
//% block="set led RGB to %color" weight=80
//% help=leds/leds_setLedRGB
//% blockId="leds_setLedRGB" blockGap=8
void setLedRGB(AllColors color) {
    // to do
}
} // namespace leds

File: pxt-steami/libs/leds/sim/leds.ts
namespace pxsim.leds {
    enum AllColors {
        RED = 0,
        GREEN = 1,
        BLUE = 2,
        YELLOW = 3,
        MAGENTA = 4,
        CYAN = 5,
        WHITE = 6,
    }

    function getHexColor(color: AllColors): string {
        switch (color) {
            case AllColors.RED:
                return '#ff0000';
            case AllColors.GREEN:
                return '#00ff00';
            case AllColors.BLUE:
                return '#0000ff';
            case AllColors.YELLOW:
                return '#ffff00';
            case AllColors.MAGENTA:
                return '#ff00ff';
            case AllColors.CYAN:
                return '#00ffff';
            case AllColors.WHITE:
                return '#ffffff';
            default:
                return '#000000';
        }
    }

    export function setLedRGBOff(): void {
        let ledState = pxsim.ledState();
        ledState.setState(6, false, '#000000', 0);
        runtime.queueDisplayUpdate();
    }

    export function setLedRGB(color: AllColors): void {
        let ledState = pxsim.ledState();
        ledState.setState(6, true, getHexColor(color), 7);
        runtime.queueDisplayUpdate();
    }
}
